app/services/facebook_oidc_auth_service.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require "jwt"
+require "net/http"
+
+class FacebookOidcAuthService < BaseAuthService
+  class << self
+    private
+
+    def primary_key
+      :facebook_user_id
+    end
+
+    def expected_fields
+      %w[user_id]
+    end
+
+    def decode_token(id_token)
+      jwks = fetch_facebook_jwks
```

Comment:
[Reuse] Could extract JWT validation logic to BaseAuthService. Evidence: Similar JWT.decode patterns in OktaAuthService line 15. Impact: Standardizes OIDC validation across services. [Category: Reuse]